server-mux.go

```diff
@@ -153,9 +167,62 @@ func (l MuxListener) Accept() (net.Conn, error) {
 	return cmux, nil
 }
 
+func (l *MuxListener) cleanup(c net.Conn) {
+	// Block until MuxServer close receives a value then cleanup listener
+	<-l.close
+
+	if c != nil {
+		c.Close()
+	}
+
+	l.close <- l.Listener.Close()
```

Comment:
By the way, there is only one Listener instance and many Conn instances, we will have to close listener only once.